__tests__/login.test.js

```diff
@@ -38,6 +38,23 @@ describe('LoginPage', () => {
       </LayoutProvider>
     );
     
-    expect(screen.getByText('Welcome, Isabel!')).toBeInTheDocument();
+    expect(screen.getByText('Pet Assistance and Welfare System')).toBeInTheDocument();
   });
-});
\ No newline at end of file
+  it('renders Sign-in Microsoft Button', () => {
+    useRouter.mockImplementation(() => ({
+      push: jest.fn(),
+    }));
+
+    render(
+      <LayoutProvider>
+        <UserContext.Provider value={{ user: null, setUser: jest.fn() }}>
+          <LoginPage />
+        </UserContext.Provider>
+      </LayoutProvider>
+    );
+    
+    expect(screen.getByLabelText('Sign in with Microsoft')).toBeInTheDocument();
+  });
+
+});
+
```

Comment:
Let's put this test in a __test__/pages/login folder (so we can find it when we have lots of tests)